src/Components/Components/src/Routing/Router.cs

```diff
@@ -69,6 +64,22 @@ public void Attach(RenderHandle renderHandle)
         public Task SetParametersAsync(ParameterView parameters)
         {
             parameters.SetParameterProperties(this);
+
+            // Found content is mandatory, because even though we could use something like <RouteView ...> as a
+            // reasonable default, if it's not declared explicitly in the template then people will have no way
+            // to discover how to customize this (e.g., to add authorization).
+            if (Found == null)
+            {
+                throw new InvalidOperationException($"The {nameof(Router)} component requires a value for the parameter {nameof(Found)}.");
+            }
+
+            // NotFound content is mandatory, because even though we could display a default message like "Not found",
+            // it has to be specified explicitly so that it can also be wrapped in a specific layout
+            if (NotFound == null)
+            {
+                throw new InvalidOperationException($"The {nameof(Router)} component requires a value for the parameter {nameof(NotFound)}.");
+            }
```

Comment:
Making these params mandatory will also help with people upgrading. If they were optional and defaulted to "display nothing", then people would just get a blank screen after upgrade.

But even ignoring upgrades, I'd also choose to make them mandatory for the reasons given in the comments.